sirepo/package_data/static/js/sirepo.js

```diff
@@ -2147,4 +2225,47 @@ SIREPO.app.controller('SimulationsController', function (appState, panelState, r
         appState.saveQuietly('simulationFolder');
     });
     loadList();
+
+    // invoked in loadList() callback
+    function checkURLForFolder() {
+
+        var pathNoSectionRE = new RegExp('^\/' + activeSection.getActiveSection());
+        var decodedPath = decodeURI($location.path());
+        var canonicalPath = compoundPathToPath(decodedPath.replace(pathNoSectionRE, ''));
+
+        if( canonicalPath !== (appState.getActiveFolderPath() || '/') ) {
+            var fList = [rootFolder()];
+            var newFolder = folderForPathInList(canonicalPath, fList);
+            if( newFolder ) {
+                appState.setActiveFolderPath(canonicalPath);
+                appState.setActiveFolder(newFolder);
+                self.openItem(newFolder);
+            }
+            else {
+                requestSender.localRedirect('notFound');
+            }
+        }
+        else {
+            if( !appState.getActiveFolder() ) {
```

Comment:
Seems like this check needs to be first. The root case would eliminate the getActiveFolderPath check (|| '/') above.